core/trace/src/main/scala/org/typelevel/otel4s/trace/Span.scala

```diff
@@ -94,6 +95,15 @@ trait Span[F[_]] extends SpanMacro[F] {
     */
   final def end(timestamp: FiniteDuration): F[Unit] =
     backend.end(timestamp)
+
+  /** Modify the context `F` using the transformation `f`. */
+  def mapK[G[_]](f: F ~> G): Span[G] = new Span.MappedK(this)(f)
```

Comment:
this could be simpler as (from @iRevive's PR)

```suggestion
  def mapK[G[_]](f: F ~> G): Span[G] = Span.fromBackend(backend.mapK(f))
```